src/main/java/com/google/api/codegen/discovery/CommonDiscoveryProvider.java

```diff
@@ -29,19 +29,23 @@
  */
 public class CommonDiscoveryProvider implements DiscoveryProvider {
   private final DiscoveryContext context;
-  private final SnippetSetRunner<Method> snippetSetRunner;
+  private final SnippetSetRunner.Generator<Method> snippetSetRunner;
   private final String snippetFileName;
 
   public CommonDiscoveryProvider(
-      DiscoveryContext context, SnippetSetRunner<Method> snippetSetRunner, String snippetFileName) {
+      DiscoveryContext context,
+      SnippetSetRunner.Generator<Method> snippetSetRunner,
+      String snippetFileName) {
     this.context = context;
     this.snippetSetRunner = snippetSetRunner;
     this.snippetFileName = snippetFileName;
   }
 
   @Override
   public Map<String, Doc> generate(Method method) {
-    GeneratedResult result = snippetSetRunner.generate(method, snippetFileName, context);
+    GeneratedResult result =
+        snippetSetRunner.generate(
+            method, SnippetSetRunner.SNIPPET_RESOURCE_ROOT, snippetFileName, context);
```

Comment:
@garrettjonesgoogle I thought about this a little more. The way that it looks right now, a Generator is essentially a function object. It holds no state, everything that affects `generate` goes through the parameter. If we put the root in the constructor, we suddenly loose this property. What do you think? I'm not strong opinionated in this case.